### ISSUE TITLE
hadolint 1.17.5 fails to install/build

### ISSUE THREAD

**Issue by bboles** [Homebrew/linuxbrew-core#19526]
# Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

New install of `hadolint`.
## What happened (include command output)
https://gist.github.com/35fa61a1df2a8b6c58242e6e43649139
## What you expected to happen
Successful install of `hadolint`.
## Step-by-step reproduction instructions (by running `brew install` commands)
`brew install hadolint`

**Comment by dawidd6** [comment]
```
optparse-applicative> <command line>: can't load .so/.DLL for: libgmp.so (libgmp.so: cannot open shared object file: No such file or directory)
```

**Comment by bboles** [comment]
> (libgmp.so: cannot open shared object file: No such file or directory)

I got past this error by just creating a softlink:
```
ln -s /usr/lib64/libgmp.so.10 /usr/lib64/libgmp.so
```
That is probably not the best way to fix this.

Got further into the build but now running into a different issue:
```
Building all executables for `hadolint' once. After a successful build of all of them, only specified executables will be rebuilt.
hadolint            > configure (lib + exe)
Configuring hadolint-1.17.4...
hadolint            > build (lib + exe)
Preprocessing library for hadolint-1.17.4..
Building library for hadolint-1.17.4..
[ 1 of 12] Compiling Hadolint.Shell
[ 2 of 12] Compiling Hadolint.Rules
[ 3 of 12] Compiling Hadolint.Formatter.Format

/tmp/hadolint-20200207-6282-17kafzo/hadolint-1.17.5/src/Hadolint/Formatter/Format.hs:16:1: warning: [-Wunused-imports]
    The import of ‘Data.Semigroup’ is redundant
    ¦ except perhaps to import instances from ‘Data.Semigroup’
    To import instances alone, use: import Data.Semigroup()
   |
16 | import Data.Semigroup
   | ^^^^^^^^^^^^^^^^^^^^^
[ 4 of 12] Compiling Hadolint.Formatter.TTY
[ 5 of 12] Compiling Hadolint.Formatter.Json
[ 6 of 12] Compiling Hadolint.Formatter.Codeclimate
[ 7 of 12] Compiling Hadolint.Formatter.Codacy
[ 8 of 12] Compiling Hadolint.Formatter.Checkstyle
[ 9 of 12] Compiling Hadolint.Lint
[10 of 12] Compiling Hadolint.Config
[11 of 12] Compiling Hadolint
[12 of 12] Compiling Paths_hadolint
Preprocessing executable 'hadolint' for hadolint-1.17.4..
Building executable 'hadolint' for hadolint-1.17.4..
[1 of 2] Compiling Paths_hadolint
[2 of 2] Compiling Main
Linking .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/hadolint/hadolint ...
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lm
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lrt
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lutil
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -ldl
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lpthread
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lm
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lrt
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -ldl
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lpthread
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lpthread
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lc
collect2: error: ld returned 1 exit status
`gcc-5' failed in phase `Linker'. (Exit code: 1)
Completed 52 action(s).

--  While building package hadolint-1.17.4 using:
    ¦ /tmp/hadolint-20200207-6282-17kafzo/hadolint-1.17.5/.brew_home/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1 build lib:hadolint exe:hadolint --ghc-options ""
    Process exited with code: ExitFailure 1
```

**Comment by bboles** [comment]
> (libgmp.so: cannot open shared object file: No such file or directory)

I was able to get past this by creating a soft link:
```
ln -s /usr/lib64/libgmp.so.10 /usr/lib64/libgmp.so
```
That is not likely the best way to fix this.

I was able to get further in the build but then got this error:
```
Building all executables for `hadolint' once. After a successful build of all of them, only specified executables will be rebuilt.
hadolint            > configure (lib + exe)
Configuring hadolint-1.17.4...
hadolint            > build (lib + exe)
Preprocessing library for hadolint-1.17.4..
Building library for hadolint-1.17.4..
[ 1 of 12] Compiling Hadolint.Shell
[ 2 of 12] Compiling Hadolint.Rules
[ 3 of 12] Compiling Hadolint.Formatter.Format

/tmp/hadolint-20200207-6282-17kafzo/hadolint-1.17.5/src/Hadolint/Formatter/Format.hs:16:1: warning: [-Wunused-imports]
    The import of ‘Data.Semigroup’ is redundant
    ¦ except perhaps to import instances from ‘Data.Semigroup’
    To import instances alone, use: import Data.Semigroup()
   |
16 | import Data.Semigroup
   | ^^^^^^^^^^^^^^^^^^^^^
[ 4 of 12] Compiling Hadolint.Formatter.TTY
[ 5 of 12] Compiling Hadolint.Formatter.Json
[ 6 of 12] Compiling Hadolint.Formatter.Codeclimate
[ 7 of 12] Compiling Hadolint.Formatter.Codacy
[ 8 of 12] Compiling Hadolint.Formatter.Checkstyle
[ 9 of 12] Compiling Hadolint.Lint
[10 of 12] Compiling Hadolint.Config
[11 of 12] Compiling Hadolint
[12 of 12] Compiling Paths_hadolint
Preprocessing executable 'hadolint' for hadolint-1.17.4..
Building executable 'hadolint' for hadolint-1.17.4..
[1 of 2] Compiling Paths_hadolint
[2 of 2] Compiling Main
Linking .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/hadolint/hadolint ...
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lm
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lrt
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lutil
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -ldl
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lpthread
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lm
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lrt
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -ldl
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lpthread
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lpthread
/home/linuxbrew/.linuxbrew/bin/ld.bfd: cannot find -lc
collect2: error: ld returned 1 exit status
`gcc-5' failed in phase `Linker'. (Exit code: 1)
Completed 52 action(s).

--  While building package hadolint-1.17.4 using:
    ¦ /tmp/hadolint-20200207-6282-17kafzo/hadolint-1.17.5/.brew_home/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1 build lib:hadolint exe:hadolint --ghc-options ""
    Process exited with code: ExitFailure 1
```

**Comment by iMichka** [comment]
Fixed by #19537

**Comment by bboles** [comment]
Installs and performs as expected now.  Thanks @iMichka and @dawidd6 !
